### PR TITLE
Marital email option bug

### LINKED ISSUE #1497: Marital email optional
## Where are you in the form?

**Section:** Relatives

**Question:** Provide your current marital/relationship status...

**Field:** Provide this person's email address

**Component (for example address block):** Email block

## What did you expect to see?

Email address for spouse is not required. 
Add "I don't know" option to remedy as specified in the validation matrix.

- [x] Validation error
- [ ] Data not saving
- [ ] Language or content issue
- [ ] Other (explain below)

## What did you see?

email address error 2
email address error 1

## Is this error limited to just this field or others in the form?

- [x] Only relevant to just this question
- [ ] Cross-form issue
- [ ] I don't know

## What operating system and browser/version are you using?
**Operating system**
- [x] Windows
- [ ] MacOS

**Browser**
- [ ] Chrome
- [ ] Firefox
- [x] IE
- [ ] Edge
- [ ] Other

**Version:** 11

### PR DESCRIPTION
**Pull request by gidjin**
## Description

Fixes #1497 Marital Email Option. Email under spouse has an I don't know checkbox.

## Checklist for Reveiwer

- [ ] Review code changes
- [ ] Review changes for Database effects
- [ ] Verify change works in IE browser

More details about this can be found in [docs/review.md](docs/review.md)

### CODE REVIEW

**Comment by christopherhuii** [comment]
The `I don't know` does not save.

If I click `I don't know` for spouse and click save, it does not save.

**Comment by ryanhofdotgov** [comment]
You will also need to make sure that a complete test scenario serializes the I don't know option to XML correctly. 

I would throw that into test case 6, see: https://github.com/18F/e-QIP-prototype/blob/develop/docs/test-scenarios.md (i.e., load up test6 with `load-scenario`, make the change in UI, serialize it back to json via `form`, use that as new test6.json, then run `make test-go`, review XML difference, put new test6.xml in place. 

Then revalidate against XSDs, then do a test submit of new test6.json to e-QIP. MacRae and I are familiar with this process.

**Comment by gidjin** [comment]
I've updated the api to save and return the "Do not know" for email under spouse. I also created a new test scenario, 10, since 6 was specifically testing divorced and this was for the married branch. I checked the XSDs and the email just has a min of 0 occurrences but does not support the DoNotKnow attribute. I've verified that the xml generated is as expected but I have not submitted against to e-QIP, I'll follow up in slack

**Comment by gidjin** [comment]
Submitted to e-QiP Test service and the test10.json file passes

**Comment by ryanhofdotgov** [comment]
It's not your fault, @gidjin but unfortunately test case 7 is the worst test case to derive a brand new test scenario from. It was developed by the NBIB team and intentionally made somewhat pathological to test the boundaries of eApp. 

I've got a brief comment to this affect on the page linked from here:
https://docs.google.com/document/d/1q8uvcc4S9gql4cr8LEJt5wy36Db75e37qzpzJYyGi-w/edit
but now I'm realizing I need to mirror that comment in `xml_test.go` and post a large "Here be dragons" sign.

Would you be able to instead redo this new test scenario so it derives from test case 1 or one of the other "smoke test" cases? That will allow it to be another mostly "blank canvas" that we can easily add tests to when it doesn't fit into test case 6.

**Comment by ryanhofdotgov** [comment]
@gidjin Another item that I'm realizing is documented [here](https://docs.google.com/document/d/1q8uvcc4S9gql4cr8LEJt5wy36Db75e37qzpzJYyGi-w/edit) but not in [test-scenarios.md](https://github.com/18F/e-QIP-prototype/blob/develop/docs/test-scenarios.md) is that all the test scenarios need to have different:
* SSN
* Birthdate
* Birthplace
* Lastname
You may have already been tracking that, but pointing it out just in case. I'll make a correction to [test-scenarios.md](https://github.com/18F/e-QIP-prototype/blob/develop/docs/test-scenarios.md

File: api/xml/xml_test.go [hunk 1]
```diff
@@ -323,6 +323,11 @@ func TestScenario9(t *testing.T) {
 	executeScenario(t, "test9")
 }
 
+// `test10` is a copy of test7 to allow testing married data without affecting OPM approved tests
```
Comment (ryanhofdotgov) [inline api/xml/xml_test.go]: Need to update this comment

File: api/testdata/complete-scenarios/test10.json [hunk 2]
```diff
@@ -0,0 +1,5748 @@
...
+                          "checked": true
+                        }
+                      },
+                      "OtherExplanation": {
+                        "type": "textarea",
+                        "props": {
+                          "value": ""
+                        }
+                      },
+                      "DocumentNumber"
```
Comment (ryanhofdotgov) [inline api/testdata/complete-scenarios/test10.json]: I think the purpose of the additional test is to have a current spouse entry where email is not applicable, if I am following correctly. So as-is, this case doesn't appear to test anything different than other complete scenarios that have a spouse.

File: api/testdata/complete-scenarios/test10.json [hunk 3]
```diff
@@ -0,0 +1,5748 @@
...
+    "Cohabitants": {
+      "type": "relationships.status.cohabitant",
+      "props": {
+        "HasCohabitant": {
+          "type": "branch",
+          "props": {
+            "value": "Yes"
+          }
+        },
+        "CohabitantList": {
```
Comment (ryanhofdotgov) [inline api/testdata/complete-scenarios/test10.json]: All these cohabitants look like the are still derived from test case 5 – test case 1 is a blank canvas in this section.

File: api/testdata/complete-scenarios/test10.json (hunk 2 above)
Comment (ryanhofdotgov) [reply api/testdata/complete-scenarios/test10.json]: I just took a glance at `spouse-present-marriage.xml`, line 42 – the XML template logic does not support the new "Not Applicable" flag in the UI. 

It needs either:
* some direct template logic like in `history-residence.xml` , line 19 **OR**
* a helper function like `monthYearOptional` (used in conjunction with conditional template logic `date-month-year-optional.xml`).

A helper function could be used in other places to make the code more defensive, e.g., `personal-references.xml`, line 12.

If you would feel more comfortable if I took on this part of the PR, let me know and we can split this PR up.

File: api/xml/xml_test.go (hunk 1 above)
Comment (gidjin) [reply api/xml/xml_test.go]: Fixed

File: api/testdata/complete-scenarios/test10.json (hunk 2 above)
Comment (gidjin) [reply api/testdata/complete-scenarios/test10.json]: I've made the updates as discussed

File: api/testdata/complete-scenarios/test10.json (hunk 3 above)
Comment (gidjin) [reply api/testdata/complete-scenarios/test10.json]: I went ahead and loaded test case 1 and then filled in the rest of the data via the ui and exported it.

**Pull request merged**